diff -Naur omniORB-4.1.4-orig/include/omniORB4/CORBA_sysdep.h omniORB-4.1.4/include/omniORB4/CORBA_sysdep.h
--- omniORB-4.1.4-orig/include/omniORB4/CORBA_sysdep.h	2011-06-28 11:24:09.798232813 -0400
+++ omniORB-4.1.4/include/omniORB4/CORBA_sysdep.h	2011-06-28 11:26:11.749481521 -0400
@@ -132,8 +132,10 @@
 // Processor dependencies
 //
 
+// __VFP_FP__ means that the floating point format in use is that of the ARM 
+// VFP unit, which is native-endian IEEE-754.
 #if defined(__arm__)
-#  if defined(__armv5teb__)
+#  if defined(__armv5teb__) || defined(__VFP_FP__)
 #    define NO_OMNI_MIXED_ENDIAN_DOUBLE
 #  else
 #    define OMNI_MIXED_ENDIAN_DOUBLE
